src/borg/_hashindex.c

```diff
@@ -1,3 +1,4 @@
+#include <Python.h>
```

Comment:
Moving it to Cython means exposing most C structures to Cython and staring at tons of generated C code to see that it actually does the right thing (like in #891). I wouldn't do it together with a functional change.